agent/taskresource/types/types_test.go

```diff
@@ -74,3 +73,17 @@ func TestMarshalUnmarshalSSMSecretResource(t *testing.T) {
 	assert.Equal(t, resourcestatus.ResourceCreated, ssmRes.GetDesiredStatus())
 	assert.Equal(t, resourcestatus.ResourceRemoved, ssmRes.GetKnownStatus())
 }
+
+func TestMarshalUnmarshalASMSecretResource(t *testing.T) {
+	bytes := []byte(`{"asmsecret":[{"TaskARN":"task_arn","RequiredSecrets":{"secret-name_us-west-2":[]},"CreatedAt":"0001-01-01T00:00:00Z","DesiredStatus":"CREATED","KnownStatus":"REMOVED"}]}`)
```

Comment:
this doesn't seem to match the actual json string that will be persisted, as specified by the json tag [in this struct]( https://github.com/yumex93/amazon-ecs-agent/blob/46eae3cbdb90a1b58caa6376eaa7f5c77815382d/agent/taskresource/asmsecret/asmsecret.go#L381). It seems that this test happens to pass because you only check CreatedAt, DesiredStatus and KnownStatus, and those three are different from the correct tags only by case, and unmarshal is case insensitive.